From 0000000000000000000000000000000000000001 Mon Sep 17 00:00:00 2001
From: l0rdg3x <gcimmino@rayonra.net>
Date: Wed, 21 May 2026 00:00:00 +0200
Subject: [PATCH] scsi: export scsi_device_from_queue unconditionally

scsi_device_from_queue() is currently exported only when
CONFIG_CDROM_PKTCDVD_MODULE is set.  Out-of-tree GPL modules that
issue SCSI Compare-And-Write (CAW, opcode 0x89) need to obtain the
scsi_device from an arbitrary request_queue to call scsi_execute_cmd().

Remove the CONFIG_CDROM_PKTCDVD_MODULE guard and retain the comment
that explains the historical reason for the export.

Signed-off-by: l0rdg3x <gcimmino@rayonra.net>
---
 drivers/scsi/scsi_lib.c | 2 --
 1 file changed, 2 deletions(-)

--- a/drivers/scsi/scsi_lib.c
+++ b/drivers/scsi/scsi_lib.c
@@ -2205,9 +2205,7 @@
  * reasons like the old IDE driver it isn't.  This export allows it to safely
  * probe if a given device is a SCSI one and only attach to that.
  */
-#ifdef CONFIG_CDROM_PKTCDVD_MODULE
 EXPORT_SYMBOL_GPL(scsi_device_from_queue);
-#endif

 /**
  * scsi_block_requests - Utility function used by low-level drivers to prevent
--
2.49.0
